### PR TITLE
[Metricbeat] Migrate Kubernetes state_replicaset Metricset to use ReporterV2 interface

### PR DESCRIPTION
**Pull request by sayden**
Refer to elastic/beats#10774 for more info

### CODE REVIEW

**Comment by sayden** [comment]
jenkins, test this

**Comment by sayden** [comment]
Error in ES module seems unrelated https://beats-ci.elastic.co/job/elastic+beats+pull-request+multijob-linux/beat=metricbeat,label=linux-immutable/5456/console

**Comment by sayden** [comment]
jenkins, test this please

File: metricbeat/module/kubernetes/state_replicaset/_meta/testdata/kube-state-metrics-v1.3.0.plain-expected.json [hunk 1]
```diff
@@ -0,0 +1,147 @@
+[
+    {
+        "event": {
+            "dataset": "kubernetes.replicaset",
+            "duration": 115000,
+            "module": "kubernetes"
+        },
+        "kubernetes": {
+            "replicaset": {
+                "_module": {
```
[the remark this answers is not in the document]
Comment (sayden) [reply metricbeat/module/kubernetes/state_replicaset/_meta/testdata/kube-state-metrics-v1.3.0.plain-expected.json]: Removed

**Comment by ruflin** [comment]
jenkins, test this

File: metricbeat/module/kubernetes/state_replicaset/state_replicaset.go [hunk 2]
```diff
@@ -91,18 +87,43 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
...
+		reporter.Error(err)
+		return
+	}
+
+	m.enricher.Enrich(events)
+	for _, event := range events {
+
+		var moduleFieldsMapStr common.MapStr
+		moduleFields, ok := event[mb.ModuleDataKey]
+		if ok {
```
Comment (ruflin) [inline metricbeat/module/kubernetes/state_replicaset/state_replicaset.go]: Can you file a follow up issue so we improve the prometheus input so we can get rid of this. I think @odacremolbap worked already on some parts.

Comment (sayden) [reply metricbeat/module/kubernetes/state_replicaset/state_replicaset.go]: Sure, I did some quick test using this https://github.com/elastic/beats/blob/master/metricbeat/module/kubernetes/state_container/state_container_test.go#L29 but it didn't work :(

Comment (sayden) [reply metricbeat/module/kubernetes/state_replicaset/state_replicaset.go]: I didn't understand why some metricsets were using it and some not

Comment (ruflin) [reply metricbeat/module/kubernetes/state_replicaset/state_replicaset.go]: Not sure I can follow your above comment. I'm ok with the current hack but want to make sure we follow up that is why the ask for an issue.

Comment (ruflin) [reply metricbeat/module/kubernetes/state_replicaset/state_replicaset.go]: As soon as we have migrated all the new interface, I think we can remove the old testing methods. I wanted to keep them around for the migration to make sure we don't miss things.

Comment (sayden) [reply metricbeat/module/kubernetes/state_replicaset/state_replicaset.go]: Issue opened https://github.com/elastic/beats/issues/11624

**Comment by ruflin** [comment]
jenkins, test this

**Comment by ruflin** [comment]
I don't think the failures are related.

**Pull request merged**